libs/wingsdk/src/core/internal.ts

```diff
@@ -62,3 +57,20 @@ ${Object.entries(clients)
 
   return new Handler();
 }
+
+function toInflight(obj: any): Code {
+  // if the object has a "_toInflight" method, use it
+  if (typeof obj === "object" && "_toInflight" in obj) {
+    return obj._toInflight();
+  }
+
+  // if this is a Set, we need to convert it to an array and reconstruct on the other side
+  if (typeof obj === "object" && (obj as any) instanceof Set) {
+    return NodeJsCode.fromInline(`new Set(${JSON.stringify(Array.from(obj))})`);
+  }
+
+  // otherwise, just JSON.stringify() it and reconstruct on the other side
+  //
+  // NOTE: this won't work if `obj` is not serializable, but this is not the ownership of the sdk
+  return NodeJsCode.fromInline(JSON.stringify(obj));
```

Comment:
Maybe worth bringing in a 3rd party lib to simplify/harden the logic? https://github.com/terrymorse58/serialize-anything

I understand wanting to avoid external deps, but this would basically solve the issue for several data types (including Duration I think). Since we don't have that many right now, I wouldn't consider it a blocker.

Reply:
I want us to be very explicit about how and what we are serializing.

We are not supposed to be serializing complex object graphs, just pure data (or resources thought their clients), so I don't see a real need for a heavy hammer.